tensorboard/webapp/metrics/store/metrics_reducers.ts

```diff
@@ -252,6 +322,40 @@ const {initialState, reducers: routeContextReducer} = createRouteContextedState(
 
 const reducer = createReducer(
   initialState,
+  // This hydration action is dispatched on any in-app navigation.
+  // switching plugin dashboards will trigger this, which clears things
+  on(stateRehydratedFromUrl, (state, {routeKind, partialState}) => {
+    if (
+      routeKind !== RouteKind.EXPERIMENT &&
+      routeKind !== RouteKind.COMPARE_EXPERIMENT
+    ) {
+      return state;
+    }
+    // The URL serializes pinned cards + pre-pinned cards. Keep these sets
+    // mutually exclusive, and do not add duplicate pre-pinned cards.
+    const seenHashes = new Set();
+    for (const pinnedCardId of state.pinnedCardToOriginal.keys()) {
+      const {tag, runId, sample} = state.cardMetadataMap[pinnedCardId];
+      seenHashes.add(hashCardUniqueInfo(tag, runId, sample));
+    }
+
+    const hydratedState = partialState as URLDeserializedState;
+    const nextUnresolvedImportedPinnedCards = [] as CardUniqueInfo[];
+    for (const card of [
+      ...state.unresolvedImportedPinnedCards,
+      ...hydratedState.metrics.pinnedCards,
```

Comment:
Would you mind leaving a small comment on why we have to read `state.unresolvedImportedPinnedCards`? I now get it that because we use routeContextedState, the unresolved cards can be from previous navigation (e.g., route A -> B -> A)

Reply:
Done